Core/DISET/RPCClient.py

```diff
@@ -3,41 +3,100 @@
 
 __RCSID__ = "$Id$"
 
+
 from DIRAC.Core.DISET.private.InnerRPCClient import InnerRPCClient
 
 class _MagicMethod( object ):
+  """ This object allows to bundle together a function calling
+      an RPC and the remote function name.
+      When this object is called (__call__), the call is performed.
+  """
+
 
   def __init__( self, doRPCFunc, remoteFuncName ):
+    """ Constructor
+
+        :param doRPCFunc: the function actually performing the RPC call
+        :param remoteFuncName: name of the remote function
+    """
     self.__doRPCFunc = doRPCFunc
     self.__remoteFuncName = remoteFuncName
 
   def __getattr__( self, remoteFuncName ):
-    return _MagicMethod( self.__doRPCFunc, "%s.%s" % ( self.__remoteFuncName, remoteFuncName ) )
+    """ I really do not understand when this would be called.
+        I can only imagine it being called by dir, or things like that.
+        In any case, it recursively return a MagicMethod object
+        where the new remote function name is the old one to which
+        we append the new called attribute.
+    """
+    return  _MagicMethod( self.__doRPCFunc, "%s.%s" % ( self.__remoteFuncName, remoteFuncName ) )
+
+
 
   def __call__(self, *args ):
+    """ Triggers the call.
+        it uses the RPC calling function given by RPCClient,
+        and gives as argument the remote function name and whatever
+        arguments given.
+    """
+
     return self.__doRPCFunc( self.__remoteFuncName, args )
 
   def __str__( self ):
     return "<RPCClient method %s>" % self.__remoteFuncName
 
 class RPCClient( object ):
+  """ This class contains the mechanism to convert normal calls to RPC calls.
+      When instanciated, it creates a :any:`~DIRAC.Core.DISET.private.InnerRPCClient.InnerRPCClient` as an attribute.
+      Any attribute which is accessed is then either redirected to InnerRPCClient if it has it,
+      or creates a MagicMethod object otherwise. If the attribute is a function, MagicMethod will
+      trigger the RPC call, using the InnerRPCClient.
+
+      The typical workflow looks like this:
+
+
+      rpc = RPCClient('DataManagement/FileCatalog')
```

Comment:
Yes it is meant to be a code block like text. Can I indent it just like I do for rst ? Or how am I supposed to do ?